pkg/topology/kademlia/kademlia.go

```diff
@@ -1260,7 +1260,7 @@ func nClosePeerInSlice(peers []swarm.Address, addr swarm.Address, spf sanctioned
 }
 
 func (k *Kad) IsReachable() bool {
-	return k.reachability == p2p.ReachabilityStatusPublic
+	return k.reachability != p2p.ReachabilityStatusPrivate
```

Comment:
this should be explicitly Public, if there is middle "unknown" state, we don't want these nodes to store chunks.